101-data-factory-v2-blob-to-blob-copy/azuredeploy.json

```diff
@@ -1,196 +1,209 @@
 {
-  "contentVersion": "1.0.0.0",
-  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
-  "parameters": {
-    "dataFactoryName": {
-      "type": "string",
-      "metadata": {
-        "description": "Name of the data factory. Must be globally unique."
-      }
-    },
-    "dataFactoryLocation": {
-      "type": "string",
-      "allowedValues": [
-        "East US",
-        "East US 2"
-      ],
-      "defaultValue": "East US",      
-      "metadata": {
-        "description": "Location of the data factory. Currently, only East US and East US 2 are supported. "
-      }
-    },
-    "storageAccountName": {
-      "type": "string",
-      "metadata": {
-        "description": "Name of the Azure storage account that contains the input/output data."
+    "$schema":"https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion":"1.0.0.0",
+    "parameters":{
+        "dataFactoryName":{
+            "type":"string",
+            "defaultValue":"GEN-UNIQUE",
+            "metadata":{
+                "description": "Data Factory Name"
+            }
+        },
+        "location":{
+            "type":"string",
+            "defaultValue":"[resourceGroup().location]",
+            "metadata":{  
+                "description":"Location of the data factory. Currently, only East US, East US 2, and West Europe are supported."
+            }
+        },
+        "storageAccountName": {
+            "type": "string",
+            "defaultValue":"GET-PREREQ-storageAccountName",
+            "metadata": {
+                "description": "Name of the Azure storage account that contains the input/output data."
       }
     },
-    "storageAccountKey": {
-      "type": "securestring",
-      "metadata": {
-        "description": "Key for the Azure storage account."
+        "storageAccountKey": {
```

Comment:
remove this since you'll use the resourceId to get the storageAccount key using listkeys() below